packages/plugin-bundle-webpack/plugin.js

```diff
@@ -0,0 +1,139 @@
+const fs = require("fs");
+const path = require("path");
+const webpack = require("webpack");
+const MiniCssExtractPlugin = require("mini-css-extract-plugin");
+const jsdom = require("jsdom");
+const CopyPlugin = require("copy-webpack-plugin");
+const { JSDOM } = jsdom;
+
+async function compilePromise(webpackConfig) {
+  const compiler = webpack(webpackConfig);
+  return new Promise((resolve, reject) => {
+    compiler.run((err, stats) => {
+      if (stats.hasErrors()) {
+        reject(err);
+      }
+      resolve(stats);
+    });
+  });
+}
+
+module.exports = function plugin(config, args) {
+  return {
+    async bundle({ srcDirectory, destDirectory, log, jsFilePaths }) {
+      let homepage = config.homepage || "";
+      let fallback = config.devOptions?.fallback || "index.html";
+
+      let dom = new JSDOM(
+        fs.readFileSync(path.join(srcDirectory, config.devOptions.fallback))
+      );
+
+      //Find the first local script, assume it's the entrypoint
+      let scriptEl;
+      for (const el of Array.from(
+        dom.window.document.querySelectorAll("script")
+      )) {
+        let src = el.src.trim();
+        if (
+          el.type.trim().toLowerCase() === "module" &&
+          !src.trim().match(/[a-zA-Z+]+:\/\//g)
+        ) {
+          scriptEl = el;
+          break;
+        }
+      }
+      if (!scriptEl) {
+        throw new Error("Can't bundle without script tag in html");
+      }
+      let entryPoint = scriptEl.src;
+
+      //Compile files using webpack
+      let webpackConfig = {
+        resolve: {
+          alias: {
+            "/web_modules": path.join(srcDirectory, "web_modules/"),
+          },
+        },
+        module: {
+          rules: [
+            {
+              test: /\.jsx?$/,
+              exclude: ["/node_modules/"],
+              use: {
+                loader: "babel-loader",
+              },
+            },
+            {
+              test: /\.s?css$/,
+              use: [
+                {
+                  loader: MiniCssExtractPlugin.loader,
+                },
+                {
+                  loader: "css-loader",
+                },
+              ],
+            },
+            {
+              test: /\.(svg|png|jpe?g|gif)$/, //We can expand this list of importable files
```

Comment:
One nice thing about how Snowpack works is that the bundle stage only needs to worry about built files, so no sass, scss, jsx, tsx, etc. 

The way our dev server works, anything that is ESM imported that isn't JS or CSS is handled via something like file-loader, so I think we can just convert this test to `.*`

Reply:
Yes that makes sense